### DISCUSSION TITLE
DKIM_signing module for outbound, DKIM module for inbound?

### DISCUSSION THREAD

**Discussion by TonyGravagno** [rspamd/rspamd#3740] [Q&A] [answered]
At the bottom of the DKIM module [doc page](https://rspamd.com/doc/modules/dkim.html) it says:

> Please use dkim_signing module for DKIM signatures. This module should not be used for these purposes any longer.

It doesn't look like there is a way to use dkim_signing to check inbound mail.
Is that message incorrect? The message doesn't say the feature has been deprecated or removed, only that it should not be used.
But 'dkim' is still one of the default-enabled C modules.
So, what module is checking DKIM signatures? If someone can clarify the actual status of these modules, I'll update the doc.

Thanks.

**Comment by fatalbanana** [comment, marked as answer]
DKIM module is still responsible for checking DKIM signatures; in the past it additionally provided support for signing but this was removed.

dkim_signing module is responsible for signing.
